src/mapgen/mg_biome.cpp

```diff
@@ -84,8 +84,72 @@ void BiomeManager::clear()
 	m_objects.resize(1);
 }
 
-////////////////////////////////////////////////////////////////////////////////
 
+// For BiomeGen type 'BiomeGenOriginal'
+float BiomeManager::getHeatAtPosOriginal(v3s16 pos, NoiseParams np_heat,
+	NoiseParams np_heat_blend, u64 seed)
+{
+	return
+		NoisePerlin2D(&np_heat,       pos.X, pos.Z, seed) +
+		NoisePerlin2D(&np_heat_blend, pos.X, pos.Z, seed);
+}
+
+
+// For BiomeGen type 'BiomeGenOriginal'
+float BiomeManager::getHumidityAtPosOriginal(v3s16 pos, NoiseParams np_humidity,
+	NoiseParams np_humidity_blend, u64 seed)
+{
+	return
+		NoisePerlin2D(&np_humidity,       pos.X, pos.Z, seed) +
+		NoisePerlin2D(&np_humidity_blend, pos.X, pos.Z, seed);
+}
+
+
+// For BiomeGen type 'BiomeGenOriginal'
+Biome *BiomeManager::getBiomeIdAtPosOriginal(v3s16 pos,
+	NoiseParams np_heat, NoiseParams np_heat_blend,
+	NoiseParams np_humidity, NoiseParams np_humidity_blend, u64 seed)
```

Comment:
`NoiseParams &np_` for each to prevent unnecessary memory copy.

Reply:
Ok.

Reply:
`const NoiseParams &np_` surely

Reply:
Will do.

Reply:
I guess because the biome heat and humidity noises shouldn't change during a game session. However i think they are changeable in-game, so i haven't added `const`.
```
* `minetest.set_mapgen_setting(name, value, [override_meta])`
   * Sets a mapgen param to `value`, and will take effect if the corresponding mapgen setting
     is not already present in map_meta.txt.
   * `override_meta` is an optional boolean (default: `false`). If this is set to true,
     the setting will become the active setting regardless of the map metafile contents.
```

Reply:
@paramat He didn't mean to define the mapgen settings to some kind of constant value. Instead, it's desired to add `const` to the function parameters, which primary indicate developers to not alter the passed noise value in that particular function.

Reply:
Ok will do.